### Importing Necessary Library

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns

%matplotlib inline

In [2]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

### Read in the dataset

In [3]:
df = pd.read_csv('Health_Dataset.csv')

### Get info about the dataset

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11582 entries, 0 to 11581
Data columns (total 97 columns):
facility_list_yn                             11582 non-null object
facility_ID                                  10674 non-null object
enumerator_name                              11582 non-null object
zone                                         11582 non-null object
state                                        11582 non-null object
lga                                          11489 non-null object
ward                                         11582 non-null object
community                                    11579 non-null object
facility_name                                11582 non-null object
facility_type                                11582 non-null object
facility_type_dontknow                       240 non-null object
management                                   11582 non-null object
management_dontknow                          170 non-null object
num_doctors_fulltime    

In [91]:
df.head()

facility_list_yn facility_ID enumerator_name           zone      state  \
0              yes        LZPU  Onwuka Clement     south_east        imo   
1              yes        OQWV      Fred Abdul  north_central       kogi   
2              yes        DDDW      Fred Abdul  north_central       kogi   
3              yes        pwep   Anthony Akpan    south_south  akwa_ibom   
4              yes        NCGG  Onwuka Clement     south_east        imo   

                lga           ward          community  \
0  imo_ohaji_egbema         Umuapu             Umuapu   
1        kogi_ibaji          Iyano           Ojuolofe   
2        kogi_ibaji           Ayah          Ojamukoje   
3    akwa_ibom_abak  Afaha Obong 1  Ikot Akpan Ikpong   
4  imo_ohaji_egbema         Umuapu             Umuapu   

                          facility_name          facility_type  \
0    St. Mary‘s Maternity Home Umuapu  primary_health_centre   
1                         PHCC Ojuolofe  primary_health_centre   
2                        Phcc Ojamukoje  primary_health_centre   
3      Health Centre, Ikot Akpan Ikpong  primary_health_centre   
4  Precious Hospital & Maternity Umuapu  primary_health_centre   

  facility_type_dontknow      management management_dontknow  \
0                    NaN  private_profit                 NaN   
1                    NaN       local_gov                 NaN   
2                    NaN       local_gov                 NaN   
3                    NaN       local_gov                 NaN   
4                    NaN  private_profit                 NaN   

   num_doctors_fulltime  num_midwives_fulltime  num_nurses_fulltime  \
0                   1.0                    1.0                  1.0   
1                   0.0                    0.0                  0.0   
2                   0.0                    0.0                  0.0   
3                   0.0                    1.0                 11.0   
4                   1.0                    1.0                  1.0   

   num_chews_fulltime num_staff_dontknow delivery_services  \
0                 0.0                NaN               yes   
1                 1.0                NaN               yes   
2                 1.0                NaN                no   
3                 1.0                NaN               yes   
4                 0.0                NaN               yes   

  delivery_services_dontknow maternity_ward maternity_ward_dontknow  \
0                        NaN            NaN                     NaN   
1                        NaN            NaN                     NaN   
2                        NaN             no                     NaN   
3                        NaN            NaN                     NaN   
4                        NaN            NaN                     NaN   

  twentyfourseven_yn  twentyfourseven_yn_dontknow antenatal_care_yn  \
0                NaN                          NaN               yes   
1                NaN                          NaN                no   
2                NaN                          NaN                no   
3                NaN                          NaN               yes   
4                NaN                          NaN               yes   

  antenatal_care_yn_dontknow family_planning_yn family_planning_yn_dontknow  \
0                        NaN                 no                         NaN   
1                        NaN                 no                         NaN   
2                        NaN                 no                         NaN   
3                        NaN                yes                         NaN   
4                        NaN                 no                         NaN   

  emergency_transport_ambulance emergency_transport_ambulance_dontknow  \
0                            no                                    NaN   
1                            no                                    NaN   
2                            no                                    Na

#### Check amount of missing values per column

In [5]:
df.isnull().sum()

facility_list_yn                                 0
facility_ID                                    908
enumerator_name                                  0
zone                                             0
state                                            0
lga                                             93
ward                                             0
community                                        3
facility_name                                    0
facility_type                                    0
facility_type_dontknow                       11342
management                                       0
management_dontknow                          11412
num_doctors_fulltime                           390
num_midwives_fulltime                          396
num_nurses_fulltime                            402
num_chews_fulltime                             392
num_staff_dontknow                           11162
delivery_services                                0
delivery_services_dontknow     

Get all columns with high number of missing values 
2k+ missing values

In [6]:
columns = df.columns
useless_columns = []

for cols in columns:
    if df[cols].isnull().sum() > 2000:
        useless_columns.append(cols)
    else:
        pass

Total number of columns with very high missing values

In [7]:
len(useless_columns)

38

In [8]:
useless_columns

['facility_type_dontknow',
 'management_dontknow',
 'num_staff_dontknow',
 'delivery_services_dontknow',
 'maternity_ward',
 'maternity_ward_dontknow',
 'twentyfourseven_yn',
 'twentyfourseven_yn_dontknow',
 'antenatal_care_yn_dontknow',
 'family_planning_yn_dontknow',
 'emergency_transport_ambulance_dontknow',
 'emergency_transport_30min',
 'emergency_transport_30min_dontknow',
 'malaria_treatment_artemisinin_dontknow',
 'measles_yn_dontknow',
 'vaccines_fridge_freezer_dontknow',
 'vaccines_transport_dontknow',
 'hiv_testing_dontknow',
 'hiv_treatment_dontknow',
 'tb_treatment_dontknow',
 'c_section_yn_dontknow',
 'improved_water_functionality',
 'improved_water_functionality_dontknow',
 'num_toilets_total',
 'num_toilets_dontknow',
 'phcn_functional',
 'phcn_functional_dontknow',
 'solar_functional',
 'solar_functional_dontknow',
 'generator_functional',
 'generator_functional_dontknow',
 'grid_proximity_if_not_connected',
 'grid_proximity_dontknow',
 'contact_phone_dontknow',
 'phot

Dealing with each useless columns one after the other to know if it is worth dropping

### Column 0

In [9]:
# Name of the column we are dealing with
print(f'Column: {useless_columns[0]}')
# Amount of available data
11582 - df[useless_columns[0]].isnull().sum()

Column: facility_type_dontknow


240

In [10]:
# Getting insight of what the data is all about
df[useless_columns[0]].value_counts().head(10)

Closed Down                     16
Facility closed down             8
CLOSED DOWN                      8
Facility Closed down             6
Facility not in use              5
Facility closed                  5
Facility has closed down         4
Private clinic                   3
THE FACILITY NO LONGER EXIST     3
Information Denied               3
Name: facility_type_dontknow, dtype: int64

Note the trend we have a common complain but put in different forms/ format

Facility closed down/CLOSED DOWN/Facility Closed down/Facility closed/Facility not in use/Facility has closed down all this means that the company has closed down, and would not operate 

In [11]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[0]].nunique()

182

We can see that out of 240 available data provided for us to work with from the *facility_type_dontknow* we have 182 unique data points. meaning the column is not totally necessary and can be dropped

### Column 1

In [12]:
# Name of the column we are dealing with
print(f'Column: {useless_columns[1]}')
# Amount of available data
print('\n',11582 - df[useless_columns[1]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[1]].value_counts().head(10))

Column: management_dontknow

 170

 Closed Down                                  16
Facility closed down                         10
Facility closed                               7
Facility Closed down                          7
Facility not in use                           5
Information Denied                            3
Facility has closed down                      3
NO LONGER EXIST                               3
Information dinied                            3
Facility locked information not available     2
Name: management_dontknow, dtype: int64


Note the trend we have a common complain but put in different forms/ format

Facility closed down/CLOSED DOWN/Facility Closed down/Facility closed/Facility not in use/Facility has closed down all this means that the company has closed down, and would not operate

In [13]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[1]].nunique()

112

### Column 2

In [14]:
# Name of the column we are dealing with
print(f'Column: {useless_columns[2]}')
# Amount of available data
print('\n',11582 - df[useless_columns[2]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[2]].value_counts().head(10))

Column: num_staff_dontknow

 420

 Facility closed down        14
Closed Down                 14
CLOSED DOWN                 12
No respondent               10
Facility not in use          7
Facility Closed down         7
Respondent not available     6
Not functional               5
Audience not granted         5
Facility closed              5
Name: num_staff_dontknow, dtype: int64


In [15]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[2]].nunique()

298

### Column 3

In [16]:
# Name of the column we are dealing with
print(f'Column: {useless_columns[3]}')
# Amount of available data
print('\n',11582 - df[useless_columns[3]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[3]].value_counts().head(10))

Column: delivery_services_dontknow

 469

 Closed Down             16
Facility not in use     14
Facility closed down    14
No respondent           13
CLOSED DOWN             12
Not functional          10
Facility Closed down     8
Facility closed          6
NO LONGER EXIST          6
Audience not granted     5
Name: delivery_services_dontknow, dtype: int64


In [17]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[3]].nunique()

303

### Column 4

In [18]:
# Name of the column we are dealing with
print(f'Column: {useless_columns[4]}')
# Amount of available data
print('\n',11582 - df[useless_columns[4]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[4]].value_counts().head(10))

Column: maternity_ward

 3300

 no     2496
yes     796
dk        8
Name: maternity_ward, dtype: int64


In [19]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[4]].nunique()

3

### Column 5

In [20]:
t_num = 5

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: maternity_ward_dontknow

 8

 Newly constructed and is yet to be commissioned. It is not in use at present    1
No staff is being currently posted here                                         1
The facility was delapedated so not functional.                                 1
not sure for now                                                                1
I am new here                                                                   1
Facility not functional                                                         1
Respondent said he did not know                                                 1
new person                                                                      1
Name: maternity_ward_dontknow, dtype: int64


In [21]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

8

### Column 6

In [22]:
t_num = 6

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: twentyfourseven_yn

 113

 no     76
yes    37
Name: twentyfourseven_yn, dtype: int64


In [23]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

2

### Column 7

In [24]:
t_num = 7

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: twentyfourseven_yn_dontknow

 0

 Series([], Name: twentyfourseven_yn_dontknow, dtype: int64)


In [25]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

0

### Column 8

In [26]:
t_num = 8

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: antenatal_care_yn_dontknow

 450

 Closed Down             16
Facility closed down    15
Facility not in use     13
No respondent           13
CLOSED DOWN             12
Not functional          10
Facility closed          7
Facility Closed down     7
NO LONGER EXIST          6
Audience not granted     6
Name: antenatal_care_yn_dontknow, dtype: int64


In [27]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

278

### Column 9

In [28]:
t_num = 9

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: family_planning_yn_dontknow

 459

 Closed Down             15
Facility closed down    15
No respondent           13
CLOSED DOWN             12
Facility not in use     12
Not functional          10
Facility closed          8
NO LONGER EXIST          7
Facility Closed down     6
No staff                 6
Name: family_planning_yn_dontknow, dtype: int64


In [29]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

281

###  Column 10

In [30]:
t_num = 10

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: emergency_transport_ambulance_dontknow

 423

 Closed Down                 16
No respondent               13
CLOSED DOWN                 12
Facility closed down        11
Not functional              10
Facility closed              9
Facility not in use          7
Facility Closed down         7
NO LONGER EXIST              7
Respondent not available     6
Name: emergency_transport_ambulance_dontknow, dtype: int64


In [31]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

255

### Column 11

In [32]:
t_num = 11

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: emergency_transport_30min

 9573

 no     6561
yes    3007
dk        5
Name: emergency_transport_30min, dtype: int64


In [33]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

3

### Column 12

In [34]:
t_num = 12

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: emergency_transport_30min_dontknow

 5

 The relations of patient have to transport patient themself              1
Matron is new here.                                                      1
The respondent does not have information about the emergency service.    1
No Staff                                                                 1
Relatives will have to look for transport                                1
Name: emergency_transport_30min_dontknow, dtype: int64


In [35]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

5

### Column 13

In [36]:
t_num = 13

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: malaria_treatment_artemisinin_dontknow

 449

 Closed Down             16
No respondent           14
Facility not in use     14
Facility closed down    13
CLOSED DOWN             12
Facility closed          9
Not functional           9
No staff                 8
NO LONGER EXIST          7
Facility Closed down     6
Name: malaria_treatment_artemisinin_dontknow, dtype: int64


In [37]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

272

### Column 14

In [38]:
t_num = 14

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: measles_yn_dontknow

 438

 Closed Down             16
No respondent           14
Facility closed down    13
Facility not in use     12
CLOSED DOWN             12
Facility closed          9
Not functional           8
NO LONGER EXIST          7
Audience not granted     6
No staff                 6
Name: measles_yn_dontknow, dtype: int64


In [39]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

265

### Column 15

In [40]:
t_num = 15

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: vaccines_fridge_freezer_dontknow

 439

 Facility closed down    17
No respondent           14
Closed Down             14
CLOSED DOWN             12
Not functional           9
Facility closed          7
NO LONGER EXIST          7
Audience not granted     6
Facility Closed down     6
Facility not in use      6
Name: vaccines_fridge_freezer_dontknow, dtype: int64


In [41]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

274

### Column 16

In [42]:
t_num = 16

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: vaccines_transport_dontknow

 436

 Facility closed down    18
Closed Down             16
No respondent           14
CLOSED DOWN             12
Not functional          11
Facility closed          9
NO LONGER EXIST          7
Facility not in use      6
Audience not granted     6
Facility Closed down     6
Name: vaccines_transport_dontknow, dtype: int64


In [43]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

258

### Column 17

In [44]:
t_num = 17

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: hiv_testing_dontknow

 440

 Closed Down             16
No respondent           15
Facility closed down    15
CLOSED DOWN             12
Not functional          12
Facility not in use      9
Facility closed          8
NO LONGER EXIST          7
Audience not granted     6
Facility Closed down     6
Name: hiv_testing_dontknow, dtype: int64


In [45]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

268

### Column 18

In [46]:
t_num = 18

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: hiv_treatment_dontknow

 438

 Facility closed down        17
Closed Down                 16
No respondent               16
CLOSED DOWN                 12
Not functional              10
Facility closed              8
Facility not in use          8
NO LONGER EXIST              7
Facility Closed down         6
Respondent not available     6
Name: hiv_treatment_dontknow, dtype: int64


In [47]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

258

### Column 19

In [48]:
t_num = 19

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: tb_treatment_dontknow

 431

 Closed Down                 16
Facility closed down        14
No respondent               14
CLOSED DOWN                 12
Facility closed              9
Not functional               9
Facility not in use          8
NO LONGER EXIST              7
No staff                     6
Respondent not available     6
Name: tb_treatment_dontknow, dtype: int64


In [49]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

255

### Column 20

In [50]:
t_num = 20

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: c_section_yn_dontknow

 426

 No respondent           15
Closed Down             15
Facility closed down    14
CLOSED DOWN             12
Not functional           9
Facility closed          9
NO LONGER EXIST          7
No staff                 6
Audience not granted     6
Facility Closed down     6
Name: c_section_yn_dontknow, dtype: int64


In [51]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

256

### Column 21

In [52]:
t_num = 21

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: improved_water_functionality

 6367

 functional        5249
not_functional    1032
under_repair        44
stolen              37
dk                   5
Name: improved_water_functionality, dtype: int64


In [53]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

5

### Column 22

In [54]:
t_num = 22

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: improved_water_functionality_dontknow

 5

 No access to information                                           1
Facility closed down                                               1
Refuse to grant interview                                          1
No respondent                                                      1
There an overhead tank but no water has been pumped into it yet    1
Name: improved_water_functionality_dontknow, dtype: int64


In [55]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

5

### Column 23

In [56]:
t_num = 23

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: num_toilets_total

 7063

 2.0     2157
1.0     1352
3.0      909
4.0      836
6.0      395
5.0      353
8.0      185
7.0      150
10.0     149
12.0      93
Name: num_toilets_total, dtype: int64


In [57]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

63

### Column 24

In [58]:
t_num = 24

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: num_toilets_dontknow

 54

 No access to information                                                                                                8
No respondent                                                                                                           2
NO ACCESS TO INFORMATION                                                                                                2
Staff does not know                                                                                                     2
No respondent to ask questions. The externals of the plumbing work made me say there is flush toilet in the facility    1
No exact figure due to the huge size of the facility                                                                    1
NO ACCESS TO INFORMATION; FACILITY NOT IN USE                                                                           1
Too many to count                                                                                             

In [59]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

44

### Column 25

In [60]:
t_num = 25

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: phcn_functional

 5346

 yes    4229
no     1094
dk       23
Name: phcn_functional, dtype: int64


In [61]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

3

### Column 26

In [62]:
t_num = 26

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: phcn_functional_dontknow

 23

 No respondent                                                                                         3
Staff are on strike                                                                                   2
facility head declined to provide information                                                         2
No access to information                                                                              1
INFORMATION NOT AVAILABLE                                                                             1
The hospital has been on strike for two weeks prior to the survey time                                1
Converted to Midwives quaters                                                                         1
The hospital has been on strike                                                                       1
Staff are strike                                                                                      1
The facility has been on

In [63]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

19

### Column 27

In [64]:
t_num = 27

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: solar_functional

 487

 yes    313
no     172
dk       2
Name: solar_functional, dtype: int64


In [65]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

3

### Column 28

In [66]:
t_num = 28

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: solar_functional_dontknow

 2

 No staff                     1
Refuse to grant interview    1
Name: solar_functional_dontknow, dtype: int64


In [67]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

2

### Column 29

In [68]:
t_num = 29

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: generator_functional

 4021

 yes    3582
no      429
dk       10
Name: generator_functional, dtype: int64


In [69]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

3

### Column 30

In [70]:
t_num = 30

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: generator_functional_dontknow

 10

 Generator has bot been used for about Two years                                              1
The hospital has been on strike for two weeks prior to the time of survey                    1
Because the generator have not been use at the facility                                      1
The facility has been on strike for the past three weeks(prior to the survey)                1
CLINIC CLOSED                                                                                1
Yet to start                                                                                 1
Refuse to grant interview                                                                    1
Generator newly donated to the health centre                                                 1
N                                                                                            1
The facility has been on strike for the past two weeks(prior to the bfi mop up exercise).    1
Name:

In [71]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

10

### Column 31

In [72]:
t_num = 31

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: grid_proximity_if_not_connected

 6236

 beyond_500m    3192
within_500m    2471
dk              573
Name: grid_proximity_if_not_connected, dtype: int64


In [73]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

3

### Column 32

In [74]:
t_num = 32

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: grid_proximity_dontknow

 573

 Xxx                                        24
Closed Down                                16
No power line in this community            15
Facility closed down                       11
Not connected                               9
No power supply in the entire community     8
CLOSED DOWN                                 8
The distance is much                        8
Facility Closed down                        7
Too far from grid substation                7
Name: grid_proximity_dontknow, dtype: int64


In [75]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

362

### Column 33

In [76]:
t_num = 33

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: contact_phone_dontknow

 1072

 No phone                        21
No network                      18
No respondent                   15
Lost phone                      12
CLOSED DOWN                     11
The Facility Has Closed Down    11
no phone                        10
He lost his phone                9
XXX                              9
No access to information         8
Name: contact_phone_dontknow, dtype: int64


In [77]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

738

### Column 34

In [78]:
t_num = 34

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: photo_signboard

 6172

 1398479882020.jpg    5
1396416668452.jpg    4
1396243688785.jpg    3
1397218621872.jpg    3
1396574766719.jpg    3
1398683759970.jpg    2
1398432924203.jpg    2
1398850987667.jpg    2
1396160734391.jpg    2
1398517881417.jpg    2
Name: photo_signboard, dtype: int64


In [79]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

6138

### Column 35

In [80]:
t_num = 35

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: photo_repairs

 5351

 1396416697254.jpg    4
1396243745198.jpg    3
1396574797420.jpg    3
1397212685657.jpg    2
1398335935680.jpg    2
1396578463504.jpg    2
1397117959053.jpg    2
1396873622571.jpg    2
1398851206565.jpg    2
1396959941310.jpg    2
Name: photo_repairs, dtype: int64


In [81]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

5330

### Column 36

In [82]:
t_num = 36

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: subscriberid

 9275

 621300250495310    86
621300250495126    84
621300129588985    81
621207430562720    72
621207610664298    64
621300250495555    63
621300064804106    57
621300250495496    53
621300250495225    51
621300038757542    51
Name: subscriberid, dtype: int64


In [83]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

745

### Column 37

In [84]:
t_num = 37

# Name of the column we are dealing with
print(f'Column: {useless_columns[t_num]}')
# Amount of available data
print('\n',11582 - df[useless_columns[t_num]].isnull().sum())

# Getting insight of what the data is all about
print('\n',df[useless_columns[t_num]].value_counts().head(10))

Column: simserial

 9383

 89234010001281021670    86
89234010001281019831    84
89234010000839704373    81
8923420034521637401     72
8923420034579653185     64
89234010001281024120    63
89234010001491122193    57
89234010001281023536    53
89234010001281020821    51
89234010001889562067    51
Name: simserial, dtype: int64


In [85]:
# checking the amount of unique values we have in the useless_column[0]
df[useless_columns[t_num]].nunique()

748